Quadriga/src/main/java/edu/asu/spring/quadriga/web/projectblog/AddProjectBlogEntryController.java

```diff
@@ -132,15 +152,51 @@ public ModelAndView addProjectBlogEntryRequest(
         if (result.hasErrors()) {
             model = new ModelAndView("sites/addprojectblogentry");
             model.getModelMap().put("project", project);
+
         } else {
             String username = principal.getName();
             IUser user = userManager.getUser(username);
             projectBlogEntry.setAuthor(user);
 
             projectBlogEntryManager.addNewProjectBlogEntry(projectBlogEntry);
             model = new ModelAndView("redirect:" + "/sites/" + project.getUnixName() + "/projectblog");
+
             model.getModelMap().put("project", project);
         }
         return model;
     }
+
+    /**
+     * This method is used to return a JSON string for visualizing a network
+     * based on the Network id selected from the UI.
+     * 
+     * @param unixName
+     *            Unix Name given for the Project
+     * @param networkId
+     *            Network Id for the network selected from the UI.
+     * @param principal
+     *            principal object which is required to fetch information about
+     *            logged in user.
+     * @param project
+     *            project instance obtained using @InjectProject annotation
+     * @return Returns a JSON string as a response entity based on the network
+     *         selected from the UI.
+     * @throws QuadrigaStorageException
+     * @throws JAXBException
+     */
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 5, userRole = {
+            RoleNames.ROLE_COLLABORATOR_OWNER, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN }) })
+    @InjectProjectByName
+    @RequestMapping(value = "sites/{projectUnixName}/visualizenetwork/{networkId}", method = RequestMethod.GET, produces = "text/plain")
+    public ResponseEntity<String> visualizeNetworks(@ProjectIdentifier @PathVariable("projectUnixName") String unixName,
+            @PathVariable("networkId") String networkId, Principal principal,
+            @CheckAccess @InjectProject IProject project) throws QuadrigaStorageException, JAXBException {
+
+        ITransformedNetwork transformedNetwork = transformationManager.getTransformedNetwork(networkId);
+        String json = null;
+        if (transformedNetwork != null) {
+            json = jsonCreator.getJson(transformedNetwork.getNodes(), transformedNetwork.getLinks());
+        }
+        return new ResponseEntity<String>(json, HttpStatus.OK);
```

Comment:
if there is no json (because there was an error retrieving a network) you shouldn't return an OK status. Also, since this is a method that doesn't return a complete page but json but is not a method of a REST controller, you need to handle exception and return proper http statuses so that your javascript can handle it appropriately.